util/collate/collate_test.go

```diff
@@ -42,22 +32,21 @@ type keyTable struct {
 	Expect []byte
 }
 
-func testCompareTable(table []compareTable, collate string, c *C) {
-	for i, t := range table {
-		comment := Commentf("%d %v %v", i, t.Left, t.Right)
-		c.Assert(GetCollator(collate).Compare(t.Left, t.Right), Equals, t.Expect, comment)
+func testCompareTable(tables []compareTable, collate string, t *testing.T) {
+	for i, table := range tables {
+		comment := Commentf("%d %v %v", i, table.Left, table.Right)
+		require.Equal(t, table.Expect, GetCollator(collate).Compare(table.Left, table.Right), comment)
 	}
 }
 
-func testKeyTable(table []keyTable, collate string, c *C) {
-	for i, t := range table {
-		comment := Commentf("%d %s", i, t.Str)
-		c.Assert(GetCollator(collate).Key(t.Str), DeepEquals, t.Expect, comment)
+func testKeyTable(tables []keyTable, collate string, t *testing.T) {
+	for i, table := range tables {
+		comment := Commentf("%d %s", i, table.Str)
+		require.True(t, reflect.DeepEqual(GetCollator(collate).Key(table.Str), table.Expect), comment)
```

Comment:
```suggestion
		require.Equal(t, table.Expect, GetCollator(collate).Key(table.Str), comment)
```

Reply:
`require.Equal` works. You can take a look at their implementation.

Reply:
Yes, they use `reflect.DeepEqual` in `ObjectsAreEqual()` function.